expenses-manager-backend/src/main/java/com/endava/expensesmanager/controller/ExpenseController.java

```diff
@@ -44,35 +45,57 @@ public ResponseEntity<ExpenseDto> updateExpense(@PathVariable int id, @RequestBo
-        BigDecimal total = expenseService.getTotalAmountByDateBetween(userId, startDate, endDate);
+    public ResponseEntity<BigDecimal> getTotalAmountByDateBetween(
+            @RequestParam int userId,
+            @RequestParam(required = false) LocalDateTime startDate,
+            @RequestParam(required = false) LocalDateTime endDate,
+            @RequestParam(required = false) String currency) {  // Include currency
+        BigDecimal total = expenseService.getTotalAmountByDateBetween(userId, startDate, endDate, currency);  // Include currency
         return ResponseEntity.ok(total);
     }
 
+
     @GetMapping("/user/{userId}/")
-    public ResponseEntity<List<ExpenseDto>> getExpensesPage(@PathVariable int userId,
-                                                            @RequestParam(required = false) LocalDateTime startDate,
-                                                            @RequestParam(required = false) LocalDateTime endDate,
-                                                            @RequestParam(defaultValue = "0") int page,
-                                                            @RequestParam(defaultValue = "10")int size,
-                                                            @RequestParam(defaultValue = "DATE") PropertyEnum sortBy,
-                                                            @RequestParam(required = false) boolean ascending,
-                                                            @RequestParam(required = false) Integer categoryId,
-                                                            @RequestParam(required = false) Integer currencyId){
-        List<ExpenseDto> expenses = expenseService.getExpensesPage(userId, startDate, endDate, page, size, sortBy, ascending, categoryId, currencyId);
+    public ResponseEntity<List<ExpenseDto>> getExpensesPage(
+            @PathVariable int userId,
+            @RequestParam(required = false) LocalDateTime startDate,
+            @RequestParam(required = false) LocalDateTime endDate,
+            @RequestParam(defaultValue = "0") int page,
+            @RequestParam(defaultValue = "10") int size,
+            @RequestParam(defaultValue = "DATE") PropertyEnum sortBy,
+            @RequestParam(required = false) boolean ascending,
+            @RequestParam(required = false) Integer categoryId,
+            @RequestParam(required = false) Integer currencyId,
+            @RequestParam(required = false) String currency) {  // Include currency
+        List<ExpenseDto> expenses = expenseService.getExpensesPage(
+                userId, startDate, endDate, page, size, sortBy, ascending, categoryId, currencyId, currency);  // Include currency
         return ResponseEntity.ok(expenses);
     }
+
+
+    @GetMapping("/convert/{expenseId}")
+    public ResponseEntity<BigDecimal> convertExpenseToCurrency(
```

Comment:
de ce este nevoie de acest endpoint? In momentul de fata cand ai nevoie de o cheltuiala intr-o anumita moneda doar trebuie sa specifici in ce moneda vrei sa iti vina rezultatul pe endpoint-ul de get